main/design_documents/ARM_NEON_SVE_BRIDGE.rst

```diff
@@ -0,0 +1,64 @@
+=======================================
+Design Document for ARM_NEON_SVE_BRIDGE
+=======================================
+
+This document describes the design of the intrinsics defined in
+("NEON_SVE bridge")<../acle.rst#neon-sve-bridge>. These C intrinsics enable the
+conversion between NEON and SVE vectors with no performance overhead, and
+duplicating NEON vectors across SVE vectors.
+
+Location
+########
+
+The intrinsics will be defined in a new header file called 
+``arm_neon_sve_bridge.h``.
+
+This header file will include ``arm_neon.h`` and ``arm_sve.h``.
+It will also define ``__ARM_NEON_SVE_BRIDGE`` as 1.
+
+The intrinsics will be defined with:
+
+.. code:: c
+
+    #ifdef __ARM_NEON_SVE_BRIDGE 1
+    //define intrinsics
+
+Suggestions & Considerations
+############################
+
+Why not enable the extraction of any 128-bit subvector in an SVE vector?
+------------------------------------------------------------------------
+
+Extracting other 128-bit subvectors would go against the no-op design.
+This functionality is already available with the ACLE intrinsic 
+``svdupq_lane``. For example:
+
+``svuint64_t sve_vector_a = ...;``
+
+``uint64x2_t neon_vec = svget_neonq_u64(svdupq_lane_u64(sve_vector_a, 1));``
+
+Why not enable the insertion of a 128-bit NEON vector into a specific SVE subvector?
+------------------------------------------------------------------------------------
+
+This behaviour is not currently available with SVE intrinsics. Such an intrinsic
+would be better placed in the SVE specification. NEON/SVE Bridge only concerns 
+bridging NEON and SVE functionality.
+
+Why not include 64-bit Vectors?
+-------------------------------
+
+Support for 64-bit vectors is not added because of low interest. The
+naming scheme is designed to cope with the possibility of supporting these in
+the future. The C standard requires a unique function name for each set of
+input and return types from these functions. C Generics enable inferring the
+intended function from its parameter types, but not from the return type.
+Therefore we adopt the NEON convention of using the letter `q` in the name to
+indicate a 128-bit vector in the return type. e.g. ``svget_neonq[_u16]``. If
+64-bit vectors are added later they will omit the `q`.
+
+Intrinsic Naming
+----------------
+
+The intrinsics will be named ``svset_neonq``, ``svget_neonq`` and
+``svdup_neonq``. These names are taken from ``svget/svset[2|3|4]``. and 
+``svdupq``. The ``q`` is from the NEON 64/128 bit naming convention.
```

Comment:
```suggestion
The intrinsics are named ``svset_neonq``, ``svget_neonq`` and
``svdup_neonq``. These names are taken from ``svget/svset[2|3|4]``. and 
``svdupq`` from the SVE ACLE specifications. The ``q`` is from the NEON 64/128 bit naming convention of the NEON intrinsics.
```
Please add a link to the specs and double check their official name:
1. SVE ACLE: https://developer.arm.com/documentation/100987/
2. NEON intrinsics: https://github.com/ARM-software/acle/tree/main/neon_intrinsics

Reply:
.rst links or github links?

Reply:
rst, the syntax should be `text <URL>`_